api/resolve/__init__.py

```diff
@@ -357,18 +362,28 @@ async def resolve_uris(
     async with Postgres.acquire() as conn:
         async with conn.transaction():
             for uri in request.uris:
-                parsed_uri = parse_uri(uri)
+                try:
+                    parsed_uri = parse_uri(uri)
+                except ValueError as e:
+                    result.append(ResolvedURIModel(uri=uri, error=str(e)))
+                    continue
+
                 if parsed_uri.project_name != current_project:
                     await conn.execute(
                         f"SET LOCAL search_path TO project_{parsed_uri.project_name}"
                     )
                     current_project = parsed_uri.project_name
-                entities = await resolve_entities(
-                    conn,
-                    parsed_uri,
-                    roots.get(current_project, {}),
-                    site_id,
-                    path_only=path_only,
-                )
+
+                try:
+                    entities = await resolve_entities(
+                        conn,
+                        parsed_uri,
+                        roots.get(current_project, {}),
+                        site_id,
+                        path_only=path_only,
+                    )
+                except ValueError as e:
+                    result.append(ResolvedURIModel(uri=uri, error=str(e)))
```

Comment:
For sake of consistency - should we still 'require' the `entities` field but force it empty? Or is that redundant data?

@Lypsolon might have opinions on that depending on how he uses the return value of `entities`?

Reply:
Yep. that would probably make sense from the compatibility point of view

Reply:
Any reason to keep this PR lingering - or are you happy @Lypsolon ?

Reply:
well tomorrow is the release for ayon usd (the target)
and tbh i really dont want to just push code and hope. 

i believe the changes are good and i would like to spend time testing it with the cpp API and see what impact it has. 
but i did not yet have the time to test what real world impact it actually makes. 

this kind of testing against changes is hopefully done via CI after the next epic